### PR TITLE
Warmup multiplication not working as expected -> Fix

### PR DESCRIPTION
**Pull request by ArneNx**
Testing out `LinearWarmup` and `ExponentialWarmup`, I noticed the strange behavior that the learning-rate did not rise during the `warmup_period`. Instead, it decreased.
Looking at the code, I noticed that the current, and not the initial learning-rate was always multiplied with the dampening-factor. This lead to the decrease.

See below my implementation to produce the behavior I expected to see.

### CODE REVIEW

**Comment by Tony-Y** [comment]
https://github.com/Tony-Y/pytorch_warmup/blob/e02ac84b6159b92c2bed82d147d8b96e7d2c8a2d/test/test_base.py#L29-L30

The current implementation expects use of a LR scheduler like above. Did you use any LR scheduler?

**Comment by ArneNx** [comment]
I'm using `MultiStepLR`. I don't see how the choice of LR scheduler would change this behavior.

**Comment by Tony-Y** [comment]
lr_scheduler.step(lr_scheduler.last_epoch+1)

Could you try this if you use pytorch 1.4 or higher?

**Comment by ArneNx** [comment]
I tried this earlier. It didn't change anything.

**Comment by Tony-Y** [comment]
What version of pytorch did you use?

**Comment by ArneNx** [comment]
It's 1.4

**Comment by Tony-Y** [comment]
There are two approaches explained in the README. Which approach did you employ?

**Comment by zhixinma** [comment]
https://github.com/Tony-Y/pytorch_warmup/blob/e99f2e5c872c97e35f587f8117706e44c00e9d3c/pytorch_warmup/base.py#L50-L53

https://github.com/Tony-Y/pytorch_warmup/blob/e99f2e5c872c97e35f587f8117706e44c00e9d3c/pytorch_warmup/base.py#L92-L93

The returned `warmup_factor` omega is always less than 1 and the operation `group['lr'] *= omega` makes it decrease step by step.

**Comment by Tony-Y** [comment]
Keep in mind that `group['lr']` must be reset by an LR scheduler in the closed (non-chainable) form before the multiplication of `omega`.

Explanation of chainable schedulers: https://github.com/pytorch/pytorch/pull/26423

**Comment by zhixinma** [comment]
Thx, that may be the point.

**Comment by Tony-Y** [comment]
The new release [v0.1.0](https://github.com/Tony-Y/pytorch_warmup/releases/tag/v0.1.0) works with PyTorch 1.4.0 or above.
